pydicom/dataset.py

```diff
@@ -1071,22 +690,37 @@ def _get_pixel_array(self):
             already_have = False
         elif self._pixel_id != id(self.PixelData):
             already_have = False
-        if not already_have and not self._is_uncompressed_transfer_syntax():
-            try:
-                # print("Pixel Data is compressed")
-                self._pixel_array = self._compressed_pixel_data_numpy()
-                self._pixel_id = id(
-                    self.PixelData
-                )  # is this guaranteed to work if memory is re-used??
-                return self._pixel_array
-            except Exception as e:
-                logger.info(
-                    "Pillow or JPLS did not support this transfer syntax")
         if not already_have:
-            self._pixel_array = self._pixel_data_numpy()
-            self._pixel_id = id(
-                self.PixelData
-            )  # is this guaranteed to work if memory is re-used??
+            last_exception = None
+            successfully_read_pixel_data = False
+            for x in [h for h in pydicom.config.image_handlers
+                      if h and h.supports_transfer_syntax(self)]:
+                try:
+                    pixel_array = x.get_pixeldata(self)
+                    self._pixel_array = self._reshape_pixel_array(pixel_array)
+                    successfully_read_pixel_data = True
+                    break
+                except Exception as e:
+                    logger.debug("Trouble with", exc_info=e)
+                    last_exception = e
+                    continue
+            if not successfully_read_pixel_data:
+                handlers_tried = " ".join(
+                    [str(x) for x in pydicom.config.image_handlers])
+                logger.info("%s did not support this transfer syntax",
+                            handlers_tried)
+                self._pixel_array = None
+                self._pixel_id = None
+                if last_exception:
+                    raise last_exception
+                else:
+                    msg = "No available image handler could " \
```

Comment:
Yes, I go back and forth on this one.  The slash is just not nice looking and will break if there is stuff after it, but it does break (rather than silently do something wrong).
The '(' blocking looks nicer, but it is also a tuple indicator and it is hard to tell:

```
msg = ('try this',
       'and this')
```
from
```
msg = ('try this'
       'and this')
```
worse, it is hard to tell which is intended.  With the slash outside the quotes, I believe the intent is always clear (is there any instance where the slash outside the quotes might mean something else even with stuff after the slash?).

all that being said, I'll do what everyone else wants to do for this...